apps/hubspot/src/components/FieldSelection.tsx

```diff
@@ -0,0 +1,121 @@
+import { Box, Checkbox, Paragraph, Text } from '@contentful/f36-components';
+import { css } from 'emotion';
+import tokens from '@contentful/f36-tokens';
+import { useMemo } from 'react';
+import { SdkField } from '../locations/Dialog';
+
+type FieldsSelectionProps = {
+  fields: SdkField[];
+  selectedFields: string[];
+  setSelectedFields: React.Dispatch<React.SetStateAction<string[]>>;
+};
+
+const FieldSelection = (props: FieldsSelectionProps) => {
+  const { fields, selectedFields, setSelectedFields } = props;
+  const supportedFields = fields.filter((field) => field.supported);
+
+  const allSelected = useMemo(() => {
+    return selectedFields.length === supportedFields.length && selectedFields.length > 0;
+  }, [selectedFields]);
+
+  const onSelectAll = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { checked } = event.target;
+    setSelectedFields(checked ? supportedFields.map((field) => field.uniqueId) : []);
+  };
+
+  return (
+    <>
+      <Paragraph lineHeight="lineHeightCondensed">
+        Select fields that you want to sync. Each field will map to a custom module that can be used
+        in Hubspot emails. If edits are made in Contentful, content will update automatically.
+      </Paragraph>
+
+      <Box
+        className={css({
+          border: `1px solid ${tokens.gray300}`,
+          borderRadius: tokens.borderRadiusSmall,
+        })}>
+        <Box
+          paddingLeft="spacingS"
+          paddingTop="spacingXs"
+          paddingBottom="spacingXs"
+          className={css({
+            borderBottom: `1px solid ${tokens.gray300}`,
+          })}>
```

Comment:
💭  The line dividing select all fields and the fields seems thicker. It may be because of this